src/components/ReportModals/CashDrawerReportModal.js

```diff
@@ -65,24 +65,20 @@ const CashDrawerReportModal = (props) => {
     value: id,
     label: name,
   }));
-  const getSources = async (value) => {
-    if (!value) {
-      return undefined;
-    }
-
-    const response = await cashDrawerReportSources.POST({ createdAt: value });
-    const responseSources = response?.sources ?? [];
-    const formatSources = responseSources.map((sourceName, index) => ({
-      value: `source-${index}`,
-      label: sourceName,
-    }));
-    setSources(formatSources || []);
 
-    return undefined;
-  };
-
-  useMemo(() => getSources(servicePointsValue), [servicePointsValue])
-    .then(response => response);
+  useMemo(() => {
+    if (servicePointsValue) {
+      cashDrawerReportSources.POST({ createdAt: servicePointsValue })
+        .then(response => {
+          const responseSources = response?.sources ?? [];
+          const formatSources = responseSources.map((sourceName, index) => ({
+            value: `source-${index}`,
+            label: sourceName,
+          }));
+          setSources(formatSources || []);
+        });
+    }
+  }, [cashDrawerReportSources, servicePointsValue]);
```

Comment:
This works correct @zburke! I feel bad that you did all this clean up work after me! I was going to discuss if lint rule ("React Hook useMemo has a missing dependency: 'getSources'. Either include it or remove the dependency array  react-hooks/exhaustive-deps") should be off. And I learned that we need to include all dependencies in its arguments.
Thanks, Zak!

Reply:
@annamelnyk, no worries. "all this clean up" was really just moving one function, and I only learned that it was necessary from reading the test and lint output. We are both learning things here!